financial-qa-search/QAReranker/__init__.py

```diff
@@ -0,0 +1,90 @@
+from typing import Dict
+import numpy as np
+import torch
+from torch.nn.functional import softmax
+from transformers import BertTokenizer, BertForSequenceClassification
+from jina.executors.rankers import Match2DocRanker
+
+class QAReranker(Match2DocRanker):
+    """
+    :class:`QAReranker` Compute QA relevancy scores using a fine-tuned BERT model.
+    """
+
+    required_keys = {"text"}
+
+    def __init__(
+            self,
+            pretrained_model_name_or_path: str,
+            model_path: str,
+            max_length: int = 512,
+            *args, **kwargs):
+        """
+        :param pretrained_model_name_or_path: Either:
+            - a string with the `shortcut name` of a pre-trained model to load from cache or download, e.g.: ``bert-base-uncased``.
+            - a string with the `identifier name` of a pre-trained model that was user-uploaded to Hugging Face S3, e.g.: ``dbmdz/bert-base-german-cased``.
+            - a path to a `directory` containing model weights saved using :func:`~transformers.PreTrainedModel.save_pretrained`, e.g.: ``./my_model_directory/``.
+            - a path or url to a `tensorflow index checkpoint file` (e.g. `./tf_model/model.ckpt.index`). In this case, ``from_tf`` should be set to True and a configuration object should be provided as ``config`` argument.
+            This loading path is slower than converting the TensorFlow
+            checkpoint in a PyTorch model using the provided conversion scripts and loading the PyTorch model afterwards.
+        :param model_name: the path where the model is stored
+        :param max_length: the max length to truncate the tokenized sequences to.
+        """
+
+        super().__init__(*args, **kwargs)
+        self.pretrained_model_name_or_path = pretrained_model_name_or_path
+        self.model_path = model_path
+        self.max_length = max_length
+
+    def post_init(self):
```

Comment:
Looks very nice, however, to handle the devices the `jina` way, I will ask you to inherit this class also from the `mixin` `TorchDevice`